File: Module 07/Lab/StarWars_Common/src/main/java/edu/neumont/dtos/DtoBase.java
package edu.neumont.dtos;

import java.beans.JavaBean;
import java.util.List;

@JavaBean
public abstract class DtoBase
{
    String url;
    String name;
    String created;
    String edited;

    public DtoBase(){}

    public String getId()
    {
        String[] parts = url.split("/");
        return parts[parts.length - 1];
    }

    public String getUrl()
    {
        return url;
    }

    public void setUrl(String url)
    {
        this.url = url;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getCreated()
    {
        return created;
    }

    public void setCreated(String created)
    {
        this.created = created;
    }

    public String getEdited()
    {
        return edited;
    }

    public void setEdited(String edited)
    {
        this.edited = edited;
    }
}

File: Module 07/dotNet/StarWars.Common/StarWars.Common/Dtos/DtoBase.cs
using System;

namespace StarWars.Common.Dtos
{
    public class DtoBase
    {
        public string Url { get; set; }
        public string Name { get; set; }
        public string Created { get; set; }
        public string Edited { get; set; }

        public int Id
        {
            get
            {
                var urlParts = Url.Split('/');
                var id = urlParts[urlParts.Length - 2];
                return int.Parse(id);
            }
        }
    }
}